### PR TITLE
(PA-5013) Add Red Hat 9 (ARM64) to the puppet_agent module task

### PR DESCRIPTION
**Pull request by skyamgarp**
- Specified 'latest' to install el-9-aarch64 from **nightlies**, since official packages not available yet.
- Updated MacOS13 first available version and use the released agent build instead of nightlies.

### CODE REVIEW

**Review by amitkarsale — APPROVED** [review]
LGTM

**Comment by skyamgarp** [comment]
@joshcooper Can you please review this?

File: task_spec/spec/acceptance/init_spec.rb [hunk 1]
```diff
@@ -69,7 +71,7 @@ def log_output_errors(result)
     # else
     # end
     case target_platform
-    when %r{osx-13}
+    when %r{el-9-aarch64}
```
Comment (bastelfreak) [inline task_spec/spec/acceptance/init_spec.rb:74]: why is this changed / why do you remove osx13 here? That change isn't reflected in the PR title

**Review by donoghuc — APPROVED** [review]
